packages/core/nuxt-theme-module/theme/pages/MyAccount/MyNewsletter.vue

```diff
@@ -38,16 +20,39 @@
 
 <script>
 import { SfTabs, SfCheckbox, SfButton, SfLink } from '@storefront-ui/vue';
+import NewsletterForm from '~/components/MyAccount/NewsletterForm';
+import { useNewsletter } from '<%= options.generate.replace.composables %>';
+import { onSSR } from '@vue-storefront/core';
+
 export default {
   name: 'MyNewsletter',
   components: {
     SfTabs,
     SfCheckbox,
     SfButton,
-    SfLink
+    SfLink,
+    NewsletterForm
   },
-  data() {
-    return { newsletter: [] };
+  setup() {
+    const { updateNewsletterData, load: loadNewsletterData, newsletter } = useNewsletter();
+
+    const updateNewsletter = async ({ newsletter, onComplete, onError }) => {
+      try {
+        const data = await updateNewsletterData(newsletter);
+        await onComplete(data);
+      } catch (error) {
+        onError(error);
+      }
```

Comment:
Unfortunately, the code you referred to is outdated. We no longer throw errors from methods. To check if the request was successful, refer to the `error` object from `useNewsletter`.